app/pages/Accounts/VerifyAddress.tsx

```diff
@@ -0,0 +1,133 @@
+import React, { useState, ReactElement, useCallback } from 'react';
+import clsx from 'clsx';
+import { useSelector } from 'react-redux';
+import ConcordiumLedgerClient from '~/features/ledger/ConcordiumLedgerClient';
+import Button from '~/cross-app-components/Button';
+import MessageModal from '~/components/MessageModal';
+import Ledger from '~/components/ledger/Ledger';
+import { asyncNoOp } from '~/utils/basicHelpers';
+import {
+    findLocalDeployedCredential,
+    hasFirstCredential,
+} from '~/utils/credentialHelper';
+import { specificIdentitySelector } from '~/features/IdentitySlice';
+import { walletIdSelector } from '~/features/WalletSlice';
+import { Account, ClassName } from '~/utils/types';
+import { useAsyncMemo } from '~/utils/hooks';
+
+interface Props extends ClassName {
+    account: Account;
+    display: ReactElement;
+}
+
+/**
+ * Allows the user to verify the address of the current account, on the ledger.
+ */
+export default function VerifyAddress({ account, display, className }: Props) {
+    const [opened, setOpened] = useState(false);
+    const [warning, setWarning] = useState<string>();
+    const identity = useSelector(specificIdentitySelector(account.identityId));
+    const walletId = useSelector(walletIdSelector);
+    const credential = useAsyncMemo(
+        () =>
+            walletId
+                ? findLocalDeployedCredential(walletId, account.address)
+                : Promise.resolve(undefined),
+        () => {},
+        [walletId, account.address]
+    );
+
+    const reasonForNotBeingVerifiable = useAsyncMemo(
+        async () => {
+            if (identity === undefined) {
+                return 'No identity was found for this account. This is an internal error that should be reported.';
+            }
+            if (!(await hasFirstCredential(account.address))) {
+                return "An account's address is derived from the initial credential on that account. The initial credential is not in this wallet, and therefore it is not possible to verify the address of the account.";
+            }
+            if (identity.version === 0) {
+                return 'The address cannot be verified, because the identity used to create this account uses a deprecated version of the key generation, which is not available on the ledger.';
```

Comment:
```suggestion
                return 'The address cannot be verified, because the identity used to create this account uses a deprecated version of the key generation, which is not available on the Ledger.';
```